src/util/lang.js

```diff
@@ -63,6 +63,17 @@ exports.delete = function (obj, key) {
 }
 
 /**
+ * Check whether the object has the property.
+ *
+ * @param {Object} obj
+ * @param {String} key
+ * @return {Boolean}
+ */
+exports.hasOwnProperty = function (obj, key) {
+  return Object.prototype.hasOwnProperty.call(obj, key)
```

Comment:
Let's cache `Object.prototype.hasOwnProperty` outside the function because this is going to be called a lot.